src/Adapters/Storage/Postgres/PostgresStorageAdapter.js

```diff
@@ -2425,9 +2428,8 @@ export class PostgresStorageAdapter implements StorageAdapter {
     return Promise.resolve();
   }
 
-  // Used for testing purposes
   async updateEstimatedCount(className: string) {
-    return this._client.none('ANALYZE $1:name', [className]);
+    return this._client.none('ANALYZE $1:name', [className]).catch(() => {});
```

Comment:
Should this really run on every obj add and delete op? What does that mean for perf when for example adding many obj, without wanting to count them, wouldn't that unnecessarily consume DB resources and slow the response time?

Reply:
An ANALYZE operation analyzes the contents of a database’s tables and collects statistics about the distribution of values in each column of every table. PostgreSQL's query engine then uses these statistics to find the best query plan.

<img width="865" alt="Screenshot 2024-07-20 at 2 56 26 AM" src="https://github.com/user-attachments/assets/2b42989f-17a4-4847-8d9f-e284b042a78f">

The estimated count code we use here with ANALYZE comes directly from the official performance guide.

> For efficiency reasons, reltuples and relpages are not updated on-the-fly, and so they usually contain somewhat out-of-date values. A VACUUM or ANALYZE operation that does not scan the entire table (which is commonly the case) will incrementally update the reltuples count on the basis of the part of the table it did scan, resulting in an approximate value. In any case, the planner will scale the values it finds in pg_class to match the current physical table size, thus obtaining a closer approximation.

https://www.postgresql.org/docs/current/planner-stats.html

Reply:
Does this PR run ANALYZE on every obj add and delete command which performs an entire table scan?

Reply:
> Does this PR run ANALYZE on every obj add and delete command which performs an entire table scan?

For large tables, ANALYZE takes a random sample of the table contents, rather than examining every row. This allows even very large tables to be analyzed in a small amount of time.

I moved the ANALYZE command to run just before the estimated count returns. The is better than full count for large datasets. We originally added estimated count because a user's dashboard was on unusable after 5 Million Rows on PG.

Reply:
The question of the perf impact still remains. If an app is currently calling count many times, even though that may not be not efficient, we run ANALYZE more often than needed. I think that queries and DB maintenance are two different realms and therefore I'm still not sure we should fuse them together like this.

Reply:
Continuing [here](https://github.com/parse-community/parse-server/pull/9220#pullrequestreview-2190014514) for better visibility.